src/Rover.cpp

```diff
@@ -94,32 +95,89 @@ void parseCommandLine(int argc, char** argv) {
 			  "LOG_WARN, LOG_ERROR)")
 		.default_value(std::string("LOG_INFO"))
 		.action([](const std::string& value) {
-			std::unordered_map<std::string, int> allowed{{"LOG_TRACE", LOG_TRACE},
-														 {"LOG_DEBUG", LOG_DEBUG},
-														 {"LOG_INFO", LOG_INFO},
-														 {"LOG_WARN", LOG_WARN},
-														 {"LOG_ERROR", LOG_ERROR}};
+			std::unordered_map<std::string, int> allowed{
+				{"LOG_ERROR", loguru::Verbosity_ERROR},
+				{"LOG_WARN", loguru::Verbosity_WARNING},
+				{"LOG_INFO", loguru::Verbosity_INFO},
+				{"LOG_DEBUG", 2},
+				{"LOG_TRACE", 1},
+			};
 
 			if (allowed.find(value) != allowed.end()) {
-				setLogLevel(allowed[value]);
-				return value;
+				loguru::g_stderr_verbosity = allowed[value];
 			}
 
 			throw std::runtime_error("Invalid log level " + value);
 		});
 
 	program.add_argument("-nc", "--no-colors")
 		.help("disables colors in console logging")
-		.action([&](const auto&) { setColorsEnabled(false); })
+		.action([&](const auto&) { loguru::g_colorlogtostderr = false; })
 		.nargs(0);
 
 	try {
+		loguru::init(argc, argv);
+		namespace fs = std::filesystem;
+
+		const auto LOG_LIFESPAN = std::chrono::hours(24 * 7);
+		try {
+			for (const auto& entry : fs::directory_iterator(fs::current_path())) {
+				if (entry.path().extension() == ".log" && entry.path().stem() != "latest") {
+					std::string dateString = entry.path().stem();
+
+					// Extract components from the date string
+					int year, month, day, hour, minute, second;
+					sscanf(dateString.c_str(), "%4d%2d%2d_%2d%2d%2d", &year, &month, &day,
+						   &hour, &minute, &second);
+
+					// Ensure scan outputs are correct
+					CHECK_EQ_F(std::stoi(dateString.substr(0, 4)), year);
+					CHECK_EQ_F(std::stoi(dateString.substr(4, 2)), month);
+					CHECK_EQ_F(std::stoi(dateString.substr(6, 2)), day);
+					CHECK_EQ_F(std::stoi(dateString.substr(9, 2)), hour);
+					CHECK_EQ_F(std::stoi(dateString.substr(11, 2)), minute);
+					CHECK_EQ_F(std::stoi(dateString.substr(13, 2)), second);
```

Comment:
This is probably fine, maybe better because it's more verbose, but you could also just check  that the return value from `sscanf` is equal to 6. @abhaybd any input on this?

Reply:
I agree, I think these `stoi` checks are redundant. I think it might be just simpler and clearer to just check that `sscanf(...) == 6`. Are there situations where `sscanf()` returns the right value but the parsed values are incorrect? I don't think so, and in fact, if the string is malformed these `stoi` calls would just throw uncaught exceptions.

Reply:
For sure. Can do!